src/main/java/duke/data/DukeData.java

```diff
@@ -0,0 +1,183 @@
+package duke.data;
+
+import duke.task.Task;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * The DukeData class manages the data of the user's task list by
+ * updating the data file in the hard disk accordingly.
+ */
+public class DukeData {
+    // every time Duke starts up, create and load a new or existing file.
+    private File _saveFile;
+    private FileWriter _fw;
+    private BufferedWriter _bw;
+    private TaskList _taskList;
+
+    /**
+     * Generates a new DukeData object which handles creation of a
+     * new txt file in the default data directory.
+     */
+    public DukeData() {
+        this._taskList = new TaskList();
+
+        // create a new File in the data file
+        File dir = new File("/Users/StudyBuddy/Desktop/CS2103/iP/duke/src/main/data");
+        try {
+            if (!dir.exists()) {
+                dir.mkdir();
+            } else {
+                String fileName = "myDukeList" + ".txt";
+                File tagFile = new File(dir, fileName);
+                try {
+                    if (!tagFile.exists()) {
+                        tagFile.createNewFile();
+                    }
+                    this._saveFile = tagFile;
+                } catch (IOException e) {
+                    System.err.println("Data file is null.");
+                }
+            }
+
+            // attach the file to FileWriter and BufferedWriter
+            this._fw = new FileWriter(this._saveFile);
+            this._bw = new BufferedWriter(this._fw);
+        } catch (IOException e) {
```

Comment:
Should not catch this exception unless you help the user recover enter a new pathname and try again.